ansible/playbooks/service/grafana.yml

```diff
@@ -0,0 +1,38 @@
+---
+
+- name: Manage Grafana
+  hosts: [ 'debops_service_grafana', 'debops_grafana' ]
+  become: True
+
+  environment: '{{ inventory__environment | d({})
+                   | combine(inventory__group_environment | d({}))
+                   | combine(inventory__host_environment  | d({})) }}'
+
+  roles:
+
+    - role: etc_services
+      tags: [ 'role::etc_services', 'skip::etc_services' ]
+      etc_services__dependent_list:
+        - '{{ grafana__etc_services__dependent_list }}'
+
+    - role: keyring
+      tags: [ 'role::keyring', 'skip::keyring', 'role::grafana' ]
+      keyring__dependent_apt_keys:
+        - '{{ grafana__keyring__dependent_apt_keys }}'
```

Comment:
You should reorder the `keyring` role to be the first one in the playbook, that way on any issues with the keyservers Ansible can stop early.

Reply:
moved on top